mlir/lib/Transforms/Utils/GreedyPatternRewriteDriver.cpp

```diff
@@ -136,6 +136,17 @@ struct DebugFingerPrints : public RewriterBase::ForwardingListener {
 };
 #endif // MLIR_ENABLE_EXPENSIVE_PATTERN_API_CHECKS
 
+#ifndef NDEBUG
+static Operation *getDumpRootOp(Operation *op) {
+  return op->getParentWithTrait<mlir::OpTrait::IsIsolatedFromAbove>();
```

Comment:
Seems to me that this would be arbitrarily large for something to be printed after each folding (think a long function...)

Reply:
Changed to dumping only the direct parent.